code/scripting/scripting.cpp

```diff
@@ -255,16 +255,47 @@ DCF(script, "Evaluates a line of scripting")
 extern char Game_current_mission_filename[];
 bool ConditionedHook::AddCondition(script_condition *sc)
 {
-	for(int i = 0; i < MAX_HOOK_CONDITIONS; i++)
+	switch (sc->condition_type)
 	{
-		if(Conditions[i].condition_type == CHC_NONE)
+	case CHC_STATE:
+		sc->condition_reference_index = gameseq_get_state_idx(sc->condition_string.c_str());
+		break;
+	case CHC_VERSION:
+		char buf[32];
```

Comment:
Careful with variable declarations in a switch-case. Please put the case into its own block.